user_sync/connector/directory_ldap.py

```diff
@@ -81,22 +78,38 @@ def __init__(self, caller_options):
         self.user_surname_formatter = LDAPValueFormatter(options['user_surname_format'])
         self.user_country_code_formatter = LDAPValueFormatter(options['user_country_code_format'])
 
-        password = caller_config.get_credential('password', options['username'])
+        auth_method = options['authentication_method'].lower()
+
+        if options['username'] is not None:
+            password = caller_config.get_credential('password', options['username'])
+        else:
+            # override authentication method to anonymous if username is not specified
+            auth_method = 'anonymous'
```

Comment:
I think it would be very helpful to users to log a message here if we are going to override the auth method.  I agree that we need this feature, but I don't want to cause confusion.  It can a simple info message like "username not specified, overriding authentication method to 'anonymous'".

Reply:
> 2019-04-09 10:22:05 98627 WARNING ldap - username not specified, overriding authentication method to 'anonymous'

Like this? Should it be warning or info?

Reply:
I think INFO is fine.  We're just trying to make users aware of this documented feature.